examples/onnxruntime/inceptionV3/inceptionv3.py

```diff
@@ -0,0 +1,236 @@
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#####################################################################################
+# Inference with ONNX Runtime
+import onnxruntime
+import time
+from torchvision import models, transforms as T
+import torch
+from PIL import Image
+import numpy as np
+import argparse
+import os
+import subprocess
+
+#Use most upto date weights
+inception_v3 = models.inception_v3(weights=models.Inception_V3_Weights.DEFAULT, progress=True)
+
+# Download ImageNet labels
+#!curl -o imagenet_classes.txt https://raw.githubusercontent.com/pytorch/hub/master/imagenet_classes.txt
+
+def parse_input_args():
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "--fp16",
+        action="store_true",
+        required=False,
+        default=False,
+        help='Perform fp16 quantization on the model before running inference',
+    )
+
+    parser.add_argument(
+        "--image_dir",
+        required=False,
+        default="../resnet50/images",
+        help='Target DIR for images to infer. Default is ./images'
+    )
+
+    parser.add_argument(
+        "--batch",
+        required=False,
+        default=1,
+        help='Batch size of images per inference',
+        type=int
+    )
+
+    parser.add_argument(
+        "--top",
+        required=False,
+        default=1,
+        help='Show top K of inference results',
+        type=int
+    )
+
+    parser.add_argument(
+        "--QPS",
+        action="store_true",
+        required=False,
+        default=False,
+        help='Show inference result in Queries-Per-Second QPS instead of inference duration (milliseconds)',
+    )
+
+    parser.add_argument(
+        "--verbose",
+        action="store_true",
+        required=False,
+        default=False,
+        help='Show verbose output',
+    )
+    
+    return parser.parse_args()
+
+def get_image_list_in_dir(dir):
+    proc = subprocess.run("ls",
+                          shell=True,
+                          stdout=subprocess.PIPE,
+                          cwd=dir)
+    fileList = proc.stdout.decode().split('\n')
+    fileList
+    return fileList
+
+def softmax(x):
+    """Compute softmax values for each sets of scores in x."""
+    e_x = np.exp(x - np.max(x))
+    return e_x / e_x.sum()
+
+def run_sample(session, categories, latency, inputs, topk, batch_size):
+    
+    io_binding = session.io_binding()
+    io_binding.bind_cpu_input('input', inputs.cpu().detach().numpy())
+    io_binding.bind_output('output')
+    start = time.time()
```

Comment:
In general when you want measure execution time, it should be done as an average over many iterations.  I think this function is doing too much, maybe just let it compute the predictions and print them, but take the benchmarking to another function that does this and also eliminates the data copy overheads?

Reply:
I think this is handled correctly now with a loop. I'm seeing a small drop off between Onnxrt and our driver about 3-10 QPS between the two